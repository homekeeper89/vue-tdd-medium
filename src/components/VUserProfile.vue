<template>
  <div class="user-profile">
<<<<<<< HEAD
    <img class="user-profile__avatar" :src="user.avatar_url">
    <div class="user-profile__name">
      {{user.name}}
    </div>
    <div class="user-profile__bio">
      {{user.bio}}
    </div>
=======
    <img class ='user-profile__avatar' :src='user.avatar_url'>
    <div class="user-profile__name">{{user.name}}</div>
    <div class="user-profile__bio">{{user.bio}}</div>
>>>>>>> feature/testing-VUserProfile-3
  </div>
</template>
<script>
export default {
  name: 'UserProfile',
  props:{
    user:{
      type:Object,
      required:true,
      default:()=>({})
    }
  }
}
</script>
