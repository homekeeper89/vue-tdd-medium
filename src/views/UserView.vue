<template>
  <div>
    <VUserSearchForm />
    <VUserProfile :user="user"/>
  </div>
</template>
<script>
import {mapState} from 'Vuex'
import VUserSearchForm from '@/components/VUserSearchForm'
import VUserProfile from '@/components/VUserProfile'
export default {
  name: 'UserView',
  components: {
    VUserSearchForm,
    VUserProfile,
  },
  computed:{
    ...mapState({
      user:'user'
    })
  }
}
</script>