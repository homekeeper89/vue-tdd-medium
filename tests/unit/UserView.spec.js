import {shallowMount} from '@vue/test-utils'
import UserView from '@/views/UserView'
import VUserSearchForm from '@/components/VUserSearchForm'
import VUserProfile from '@/components/VUserProfile'

describe('UserView', () => {
  const build = ()=>{
    const wrapper = shallowMount(UserView)
    return {
      wrapper,
      userProfile : ()=>wrapper.find(VUserProfile),
      userSearchForm : ()=>wrapper.find(VUserSearchForm)
    }
  }

  it('render main wrapper ', ()=>{
    const {wrapper} = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('render child component from wrapper', ()=>{
    const {userProfile, userSearchForm} = build()
    expect(userProfile().exists()).toBe(true)
    expect(userSearchForm().exists()).toBe(true)
  })

  it('pass data from parent to child', ()=>{
    const {wrapper, userProfile} = build()
    wrapper.setData({
      user:{
        name:'Daniel'
      }
    })
    expect(userProfile().vm.user).toBe(wrapper.vm.user)
  })
})