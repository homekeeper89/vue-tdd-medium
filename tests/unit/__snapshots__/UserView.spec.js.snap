// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView render main wrapper  1`] = `
<div>
  <vusersearchform-stub></vusersearchform-stub>
  <vuserprofile-stub user="[object Object]"></vuserprofile-stub>
</div>
`;
